src/hotspot/share/opto/phaseX.cpp

```diff
@@ -2058,6 +2060,29 @@ bool PhaseIterGVN::verify_Identity_for(Node* n) {
   tty->print_cr("%s", ss.as_string());
   return true;
 }
+
+// Some other verifications that are no specific to a particular transformation
+bool PhaseIterGVN::verify_node_invariants_for(Node* n) {
+  if (n->is_AddP()) {
+    Node* addp = n->in(AddPNode::Address);
+    if (!addp->is_AddP()) {
+      return false;
+    }
+    if (addp->in(AddPNode::Base)->is_top()) {
+      return false;
+    }
+    if (addp->in(AddPNode::Base) == n->in(AddPNode::Base)) {
+      return false;
+    }
```

Comment:
```suggestion
    if (!addp->is_AddP() ||
        addp->in(AddPNode::Base)->is_top() ||
        addp->in(AddPNode::Base) == n->in(AddPNode::Base)) {
      return false;
    }
```

Reply:
It could be a bit compacted.

How do you imagine `verify_node_invariants_for` will grow over time? I suspect it will become a laundry-list of invariants, we continue going down through it as long as no invariant is violated. For that, it may make more sense to invert your condition, and assert/print inside the if-block. It would make the code more extendable.